.github/workflows/ci_cd.yml

```diff
@@ -10,7 +10,7 @@ on:
   merge_group:
 
 env:
-  MAIN_PYTHON_VERSION: '3.9'
+  MAIN_PYTHON_VERSION: '3.12'
```

Comment:
Other pyansys packages tend to use the oldest supported Python package for this, as far as I can tell. We should be consistent.

```suggestion
  MAIN_PYTHON_VERSION: '3.10'
```

Reply:
Hmm, I made this change on all our packages...

Reply:
https://github.com/ansys/grantami-recordlists/blob/main/.github/workflows/ci_cd.yml
https://github.com/ansys/grantami-jobqueue/blob/main/.github/workflows/ci_cd.yml
https://github.com/ansys/grantami-serverapi-openapi/blob/main/.github/workflows/build_and_test_library.yml
https://github.com/ansys/grantami-bomanalytics-openapi/blob/main/.github/workflows/build_and_test_library.yml

Reply:
I had a look at some other repos, e.g. [pymapdl](https://github.com/ansys/pymapdl/blob/ededf7320da0594b4b050ddeec630666c403069e/.github/workflows/ci.yml#L23), and there they're using 3.11. Maybe there is no convention, in which case this is fine.